scipy/optimize/tests/test_optimize.py

```diff
@@ -202,6 +202,40 @@ def test_bfgs(self):
                          [0, -5.24885582e-01, 4.87530347e-01]],
                         atol=1e-14, rtol=1e-7)
 
+    def test_bfgs_dfo(self):
+        # Broyden-Fletcher-Goldfarb-Shanno optimization routine
+        if self.use_wrapper:
+            opts = {'maxiter': self.maxiter, 'disp': self.disp,
+                    'return_all': False}
+            res = optimize.minimize(self.func, self.startparams,
+                                    jac=None, method='BFGS', args=(),
+                                    options=opts)
+
+            params, fopt, gopt, Hopt, func_calls, grad_calls, warnflag = (
+                    res['x'], res['fun'], res['jac'], res['hess_inv'],
+                    res['nfev'], res['njev'], res['status'])
+        else:
+            retval = optimize.fmin_bfgs(self.func, self.startparams, fprime=None,
+                                        args=(), maxiter=self.maxiter,
+                                        full_output=True, disp=self.disp,
+                                        retall=False)
+            (params, fopt, gopt, Hopt,
+             func_calls, grad_calls, warnflag) = retval
+
+        assert_allclose(self.func(params), self.func(self.solution),
+                        atol=1e-6)
+        
+        # Ensure that function call counts are 'known good'; these are from
+        # SciPy 1.21.1. Don't allow them to increase.
+        assert_(self.funccalls == 58, self.funccalls) 
+        assert_(self.gradcalls == 0, self.gradcalls) 
```

Comment:
I think we have a modest preference for using plain `assert` with `pytest` these days

Reply:
I used `assert_` just to be consistent with the rest of the file, but I see the discussion on preferring `assert` in new code...would it be helpful for me to change that to `assert` for the whole file, or just my added test?

Reply:
I just made a separate PR to replace `assert_` with `assert` in `test_optimize.py` and `test_linesearch.py`.

Reply:
Thanks, that PR is merged now.